File: DevPlaceChallenges.c
void challenge_1() {
    int numero = 0;
    scanf("Ingrese un numero ", &numero);
    do {
        if (esPrimo(numero))
            printf("Es primo");
        else
            printf("NO es primo");
        scanf("Ingrese un numero ", &numero);
    }
    while (numero!=0);
}

int esPrimo(int numero){
    for (int i = 2; i < numero; i++)
        if (numero % i === 0)
            return 0;
    return 1;
}

void challenge_2() {
    /*
    * Validacion de contrasenia con 3 intentos posibles
    */
    char[31] pass = "";
    scanf("Ingrese una contrasenia ", pass);
    char[31] newPass = "";
    int acierto = 0;
    int intentosFallidos = 0;

    while (intentosFallidos < 3 && !acierto) {
        scanf("Ingrese una contrasenia ", newPass);
        if (strcmp(newPass,pass)==0) {
            printf("Contrasenia erronea, intenta nuevamente");
            intentosFallidos++;
        }
        else
            acierto = 1;
    }

    if (acierto)
        printf("Felicitaciones, recordas la contrasenia!");
    else
        printf("Tenes que ejercitar la memoria");
}

void challenge_3() {
    float valorXHora;
    char[24] nombre = "";
    int antiguedad;
    int cantHorasTrabajadas;
    float importe;

    scanf("\n Ingrese el valor x hora: %f", &valorXHora);
    scanf("\n Ingrese el nombre del empleado: %s", nombre);
    scanf("\n Ingrese la antiguedad: %d", &antiguedad);
    scanf("\n Ingrese la cantidad de horas trabajadas: %d", &cantHorasTrabajadas);
    importe = cantHorasTrabajadas * valorXHora;

    if (antiguedad>10) 
        importe+= antiguedad * 30;
    printf("Nombre: %s \t Antiguedad: %d \t Importe a cobrar: $%.2f",nombre,antiguedad,importe);
}

void challenge_4() {
    int guess;
    int random = rand() % 1001;

    printf("\n Ingrese un numero ");
    scanf("%d", &guess);
    do {
        if (guess > random)
            printf("\n Es menor");
        else
            printf("\n Es mayor");
        printf("\nIngrese un numero ");
        scanf("%d", &guess);
    } while (guess != random);
    printf("\n Numero %d encontrado!",random);
}

void challenge_5() {
    int mayor = -999, menor = 999, sum = 0, sumPares = 0, numero;
    
    printf("\nIngrese un numero ");
    scanf("%d", &numero); 
    do {
        if (numero > mayor) 
            mayor = numero;
        if (numero < menor)
            menor = numero;
        if (numero % 2 == 0)
            sumPares+= numero;
        sum+= numero;
        printf("\nIngrese un numero ");
        scanf("%d", &numero); 
    } while (numero != -1);
    printf("Mayor: %d \t Menor: %d \t Suma: %d \t Suma Pares: %d",mayor,menor,sum,sumPares);
}

File: DevPlaceChallenges.js
const challenge_1 = () => {
    /*
     * Determinacion de si un numero ingresado es o no es primo
     */
    let numero = prompt("Ingrese un numero");
    do {
        if (esPrimo(numero))
            console.log("Es primo");
        else
            console.log("NO es primo");
        numero = prompt("Ingrese un numero");
    }
    while (numero !== 0);
}

const esPrimo = numero => {
    for (let i = 2; i < numero; i++)
        if (numero % i === 0)
            return false;
    return true;
}

const challenge_2 = () => {
    /*
    * Validacion de contrasenia con 3 intentos posibles
    */
    let pass = prompt("Ingrese una contrasenia");
    let newPass = "";
    let acierto = false;
    let intentosFallidos = 0;

    while (intentosFallidos < 3 && !acierto) {
        newPass = prompt("Ingrese nuevamente la contrasenia");
        if (newPass !== pass) {
            console.log("contrasenia erronea, intenta nuevamente");
            intentosFallidos++;
        }
        else
            acierto = true;
    }

    if (acierto)
        console.log("Felicitaciones, recordas la contrasenia!");
    else
        console.log("Tenes que ejercitar la memoria");
}

const challenge_3 = () => {
    /*
    * Calculo del importe a cobrar por un empleado a partir de datos
    * de valor por hora, horas trabajadas y antiguedad
    * 
    */
    const valorXHora = prompt("Ingrese el valor x hora de un empleado");
    
    const nombre = prompt("Ingrese el nombre del empleado");
    
    const antiguedad = prompt("Ingrese la antiguedad del empleado");
    
    const cantHorasTrabajadas = prompt("Ingrese la cantidad de horas trabajadas");
    
    let importe = cantHorasTrabajadas * valorXHora;		

    if (antiguedad>10)
        importe += antiguedad * 30;        
    console.log(`Nombre: ${nombre} Antiguedad: ${antiguedad} Importe a cobrar: $${importe}`);
}

const challenge_4 = () => {
    /*
    * Adivinanza de un numero aleatorio entre 0 y 1000, con pista de si el  
    * numero ingresado es mayor, menor o igual al buscado
    */
    const random = Math.floor(Math.random() * 1001);
    console.log("Ingrese un numero");
    let guess = prompt("Ingrese un numero"); 

    do {    
        if (guess>random)
            console.log("Es menor");
        else
            console.log("Es mayor");
        guess = prompt("Ingrese un numero"); //nuevo numero
    } while (guess !== random);

    console.log(`Numero ${random} encontrado!`);
}

const challenge_5= () => {
    /*
     * ingreso de numeros que seran registrados como el mayor, menor y
     * la suma de todos los numeros ingresados
     */
    let mayor = -999;
    let menor = 999;
    let sum = 0;
    let sumPares = 0;
    
    console.log("Ingrese un numero");
    let numero = prompt("Ingrese un numero"); 
    do {
        if (numero > mayor) 
            mayor = numero;
        if (numero < menor)
            menor = numero;
        if (numero % 2 == 0)
            sumPares+= numero;
        sum+= numero;
        numero = prompt("Ingrese un numero");
    } while (numero !== -1);
    console.log(`Mayor: ${mayor} Menor: ${menor} Suma: ${sum} Suma Pares: ${sumPares}`);
}

